content/en/docs/Reference/kairos-factory.md

```diff
@@ -230,31 +203,100 @@ Extra services can be either added or made required, etc..
 
 The separation it's also very useful for caching, as once the install phase has been cached by docker we can modify the following steps to fix any issues before the init phase is run, without removing the cache. This is particularly useful under cross platform builds where speed can take a big impact depending on your setup. 
 
-## Examples
+## Skipping steps and stages
 
-All the examples are using the [kairos default dockerfile](https://github.com/kairos-io/kairos/blob/master/images/Dockerfile) as the base Dockerfile and its reproduced below:
+We recognize that one size does not fit all, and sometimes you may want to skip certain steps or stages during the image build process. To accommodate this, kairos-init provides the `--skip-steps` flag. 
 
-```Dockerfile
-ARG BASE_IMAGE=ubuntu:20.04
+This is useful if you want to customize the image yourself and find that some steps collide with your customization. You can choose between `install` and `init` to skip those full stages or go into specific steps.
+
+Run `kairos-init steps-info` to see the available steps and their descriptions. You can pass more than one step, separated by comma, to skip multiple steps, for example: `--skip-steps installPackages,kernel`.
+
+
+## Extending stages with custom actions
+
+This allows to load stage extensions from a dir in the filesystem to expand the default stages with custom logic.
+
+You can enable this feature by using the `-x` flag
+
+The structure is as follows:
+
+We got a base dir which is `/etc/kairos-init/stage-extensions` (this is the default, but you can override it using the `KAIROS_INIT_STAGE_EXTENSIONS_DIR` env var)
+
+You can drop your custom [yip files](https://github.com/mudler/yip) and as usual, they will be loaded and executed in lexicographic order.
+
+So for example, if we have:
+- /etc/kairos-init/stage-extensions/10-foo.yaml
+- /etc/kairos-init/stage-extensions/20-bar.yaml
+- /etc/kairos-init/stage-extensions/30-baz.yaml
+
+The files will be loaded in the following order:
+- 10-foo.yaml
+- 20-bar.yaml
+- 30-baz.yaml
 
-FROM quay.io/kairos/kairos-init:v0.2.6 AS kairos-init
+The files are loaded using the yip library, so you can use all the features of [yip](https://github.com/mudler/yip) to expand the stages.
 
-FROM ${BASE_IMAGE} AS base-kairos
-ARG VARIANT=core
-ARG MODEL=generic
-ARG TRUSTED_BOOT=false
-ARG KUBERNETES_DISTRO=k3s
-ARG KUBERNETES_VERSION=latest
-ARG FRAMEWORK_VERSION=v2.16.1
-ARG VERSION
+The current stages available are:
+- before-install: Good for adding extra repos and such.
+- install: Good for installing packages and such.
+- after-install: Do some cleanup of packages, add extra packages, add different kernels and remove the kairos default one, etc.
```

Comment:
oh interesting, so you can either extend the dockerfile, or do it here to add packages, didn't know